scripts/refresh-tokens.py

```diff
@@ -1,64 +1,121 @@
+    return a dictionary of role name and credentials, e.g.
+    """
+    sts = s.client('sts')
+    results = dict()
+    for rname, arn in roles.items():
+        results[rname] = sts.assume_role(RoleArn=arn, RoleSessionName=f"windmill-{rname}")
+    return results
+
+
+def get_likely_iam_roles(prefix: str, session: boto3.session.Session) -> Dict[str, dict]:
     """Role names should be prefixed by the :prefix: argument.
     The IAM role can have the filed names corresponding to the windmill workspace, and a final group.
     Since it's possible for groups and workspaces to have various charcters, the convention will be that
     the workspace is determined by w=<workspace name> and the group by g=<group name>.
 
     So if the windmill IAM role space starts with is ``windmill-stage`` and we have a workspace called ``infra`` and a group called
     ``aws-ro`` the IAM role that would be matched with that, if it exists, would be called
-    ``/windmill-stage/w=infra/g=aws-ro``
+    ``ec2-ro``
+    and it would be in the namespace:
+    ``/windmill-stage/w=infra/``
 
     """
     iam_c = session.client('iam')
     marker = None
-    all_matching_role_names = list()
+    workspace_roles = dict()
     while True:
         if marker:
             result = iam_c.list_roles(PathPrefix=prefix, Marker=marker)
         else:
             result = iam_c.list_roles(PathPrefix=prefix)
         for r in result['Roles']:
-            all_matching_role_names.append({r['RoleName']: r['Arn']})
+            pth = r['Path']
+            wspace = pth.split('/')[2][2:]  # turn /foo/w=<something>/ into <something>
+            if not workspace_roles.get(wspace):
+                workspace_roles[wspace] = dict()
+            workspace_roles[wspace][r['RoleName']] = r['Arn']
 
         if result['IsTruncated'] is False:
             break
         marker = result['Marker']
+    return workspace_roles
 
 
+def set_ws_folder(ws_name: str, temp_creds: dict):
+    folders = _list_folders(ws_name)
+    if "aws" not in folders:
+        _create_folder(ws_name)
+    _update_folder(ws_name, temp_creds)
+
+def get_wmill_workspace_names() -> set:
+    """Provide a set of workspace names that really exist"""
+    workspace_names = set()
+    for wspace in list_workspaces_as_super_admin.sync(client=wmill.create_client()):
+        workspace_names.add(wspace.id)
+    return workspace_names
+
+
+def _list_folders(workspace):
+    """List the folders that exist in the requested workspace"""
+    res = list_folders.sync(workspace=workspace, client=wmill.create_client())
+    return res
+
+
+def _update_folder(workspace: str, content: dict):
+    """content should be a dictionary, with sub-dicts of <role
+    name>:<credentials> for each of the IAM roles that will be placed
+    in the folder.
+    """
+    request = None  # XXX I'm unsure what adds content to the folder based on the openapi
```

Comment:
Update seems to update the owners or permissions, based on https://github.com/windmill-labs/windmill/blob/main/backend/windmill-api/openapi.yaml#L3922-L3951. I'm unclear on what puts content into the folder.